pyuvdata/uvcal/uvcal.py

```diff
@@ -3351,6 +3356,436 @@ def _convert_to_filetype(self, filetype):
             setattr(other_obj, p, param)
         return other_obj
 
+    def initialize_from_uvdata(
+        self,
+        uvdata,
+        gain_convention,
+        cal_style,
+        future_array_shapes=True,
+        metadata_only=True,
+        include_uvdata_history=True,
+        cal_type="gain",
+        times=None,
+        integration_time=None,
+        time_range=None,
+        frequencies=None,
+        channel_width=None,
+        flex_spw=None,
+        flex_spw_id_array=None,
+        wide_band=None,
+        freq_range=None,
+        spw_array=None,
+        jones=None,
+        ref_antenna_name=None,
+        sky_catalog=None,
+        sky_field=None,
+        diffuse_model=None,
+        baseline_range=None,
+        Nsources=None,  # noqa
+        observer=None,
+        gain_scale=None,
+        git_hash_cal=None,
+        git_origin_cal=None,
+        extra_keywords=None,
+    ):
+        """
+        Initialize this object based on a UVData object.
+
+        Parameters
+        ----------
+        uvdata : UVData object
+            The UVData object to initialize from.
+        gain_convention : str
+            What gain convention the UVCal object should be initialized to
+            ("minus" or "plus").
+        cal_style : str
+            What calibration style the UVCal object should be initialized to
+            ("sky" or "redundant").
+        future_array_shapes : bool
```

Comment:
That's an interesting idea, it probably would make things simpler. I'll check with the core devel team about what they think.

Reply:
We want to keep it this way because we want to nudge users to use the future array shapes on UVCal, even if their UVData object doesn't use the future shapes.